mobile/android/genymotion/genymotion-service.ts

```diff
@@ -0,0 +1,152 @@
+import { AndroidVirtualDevice, DeviceTypes, NOT_RUNNING_EMULATOR_STATUS } from "../../../constants";
+import { EOL } from "os";
+
+export class AndroidGenymotionService implements Mobile.IAndroidVirtualDeviceService {
+	constructor(private $adb: Mobile.IAndroidDebugBridge,
+		private $childProcess: IChildProcess,
+		private $devicePlatformsConstants: Mobile.IDevicePlatformsConstants,
+		private $emulatorHelper: Mobile.IEmulatorHelper,
+		private $fs: IFileSystem,
+		private $hostInfo: IHostInfo,
+		private $logger: ILogger,
+		private $virtualBoxService: Mobile.IVirtualBoxService) { }
+
+	public async getAvailableEmulators(adbDevicesOutput: string[]): Promise<Mobile.IAvailableEmulatorsOutput> {
+		const availableEmulatorsOutput = await this.getAvailableEmulatorsCore();
+		const genies = availableEmulatorsOutput.devices;
+		const runningEmulatorIds = await this.getRunningEmulatorIds(adbDevicesOutput);
+		const runningEmulators = await Promise.all(runningEmulatorIds.map(emulatorId => this.getRunningEmulatorData(emulatorId, genies)));
```

Comment:
In case one of the action fails, we'll not get data for any of the Genymotion emulators. Consider using [settlePromises](https://github.com/telerik/mobile-cli-lib/blob/2bdc7a2b87e78dcc71e248802fc1b9d0869585de/helpers.ts#L155)